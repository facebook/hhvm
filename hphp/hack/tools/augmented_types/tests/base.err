hphp/hack/tools/augmented_types/tests/base.php:15:25 function/method index_by param $b does not convert cleanly: Hack doesn't have a generic object type
hphp/hack/tools/augmented_types/tests/base.php:21:18 function/method more_stuff param $a has no type
hphp/hack/tools/augmented_types/tests/base.php:21:18 function/method more_stuff has no return type
hphp/hack/tools/augmented_types/tests/base.php:24:18 function/method no_docblock has no AT docblock
