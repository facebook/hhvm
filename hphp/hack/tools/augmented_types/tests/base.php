<?php

/**
 * Some nice class.
 */
class NiceClass {

  /**
   * Does some stuff.
   *
   * @param array $a
   * @param string|int|object|bool $b
   * @return array
   */
  public static function index_by($a, $b) {
  }

  /**
   * Does some more stuff.
   */
  public function more_stuff($a) {
  }

  public function no_docblock() {
  }
}
