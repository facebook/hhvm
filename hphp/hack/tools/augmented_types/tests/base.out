<?php

/**
 * Some nice class.
 */
class NiceClass {

  /**
   * Does some stuff.
   *
   * @param array $a
   * @param string|int|object|bool $b
   * @return array
   */
  public static function index_by(array $a, $b): array {
  }

  /**
   * Does some more stuff.
   */
  public function more_stuff($a) {
  }

  public function no_docblock() {
  }
}
